test/unit/integrated_function_test.C

```diff
@@ -0,0 +1,309 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// GRINS - General Reacting Incompressible Navier-Stokes
+//
+// Copyright (C) 2014-2016 Paul T. Bauman, Roy H. Stogner
+// Copyright (C) 2010-2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#include "grins_config.h"
+
+#ifdef GRINS_HAVE_CPPUNIT
+
+#include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/TestCase.h>
+
+#include "test_comm.h"
+#include "grins_test_paths.h"
+#include "regression_helper.h"
+
+// GRINS
+#include "grins/math_constants.h"
+#include "grins/grins_enums.h"
+#include "grins/integrated_function.h"
+#include "grins/composite_qoi.h"
+#include "grins/simulation_builder.h"
+#include "grins/simulation.h"
+#include "grins/variable_warehouse.h"
+
+// libMesh
+#include "libmesh/parsed_function.h"
+#include "libmesh/qoi_set.h"
+#include "libmesh/steady_solver.h"
+#include "libmesh/mesh_refinement.h"
+#include "libmesh/elem.h"
+
+namespace GRINSTesting
+{
+  class IntegratedFunctionTest : public CppUnit::TestCase,
+                                 public RegressionHelper
+  {
+  public:
+    CPPUNIT_TEST_SUITE( IntegratedFunctionTest );
+
+    CPPUNIT_TEST( test_exact_answer );
+    CPPUNIT_TEST( test_convergence );
+
+    CPPUNIT_TEST_SUITE_END();
+
+  public:
+
+    void setUp()
+      {
```

Comment:
I took the stuff from here and moved it to `init_sim()` because I needed the `filename` parameter since the QoI test uses a different input file. Squashed in deleting it